test/NCI.OCPL.Api.CTSListingPages.Tests/Tests/Services/ESLabelLookupQueryServiceTest.Get.cs

```diff
@@ -0,0 +1,165 @@
+        /// Test to verify that Elasticsearch requests are being assembled correctly.
+        /// </summary>
+        [Fact]
+        public async void Get_TestRequestSetup()
+        {
+            const string theName = "supportive-care";
+            JObject expectedRequest = JObject.Parse(
+@"{
+    ""query"": {
+        ""bool"": {
+            ""should"": [
+                { ""term"": { ""pretty_url_name"": { ""value"": ""supportive-care"" } } },
+                { ""term"": { ""id_string"":       { ""value"": ""supportive-care"" } } }
+            ]
+        }
+    }
+}
+");
+
+            Uri esURI = null;
+            string esContentType = String.Empty;
+            HttpMethod esMethod = HttpMethod.DELETE; // Basically, something other than the expected value.
+
+            JObject requestBody = null;
+
+            ElasticsearchInterceptingConnection conn = new ElasticsearchInterceptingConnection();
+            conn.RegisterRequestHandlerForType<Nest.SearchResponse<LabelInformation>>((req, res) =>
+            {
+                // We don't really care about the response for this test.
+                res.Stream = MockEmptyResponse;
+                res.StatusCode = 200;
+
+                esURI = req.Uri;
+                esContentType = req.ContentType;
+                esMethod = req.Method;
+                requestBody = conn.GetRequestPost(req);
+            });
+
+            // The URI does not matter, an InMemoryConnection never requests from the server.
+            var pool = new SingleNodeConnectionPool(new Uri("http://localhost:9200"));
+
+            var connectionSettings = new ConnectionSettings(pool, conn);
+            IElasticClient client = new ElasticClient(connectionSettings);
+
+            // Setup the mocked Options
+            IOptions<ListingPageAPIOptions> clientOptions = GetMockOptions();
+
+            ESLabelLookupQueryService query = new ESLabelLookupQueryService(client, clientOptions, new NullLogger<ESLabelLookupQueryService>());
+
+            // For this test, we don't really care that this returns anything, only that the intercepting connection
+            // sets up the request correctly.
+            await query.Get(theName);
+
+            Assert.Equal("/labelinformationv1/LabelInformation/_search", esURI.AbsolutePath);
+            Assert.Equal("application/json", esContentType);
+            Assert.Equal(HttpMethod.POST, esMethod);
+            Assert.Equal(expectedRequest, requestBody, new JTokenEqualityComparer());
+        }
+
+        public static IEnumerable<object[]> Get_Scenarios = new[]
+        {
+            new object[] { new Get_NoResults() },
+            new object[] { new Get_SingleResult() }
+        };
+
+        /// <summary>
+        /// Test to verify handling when no results are returned.
+        /// </summary>
+        [Theory, MemberData(nameof(Get_Scenarios))]
+        public async void Get_TestNoResultsFound(Get_BaseScenario data)
+        {
```

Comment:
This is confusing.  The test name says "_No_ results found", but the member data includes a case where you're expecting a single document to be returned.   Is this a bad method name?